### PR TITLE
change ESCAPE from flag to bool

### PR DESCRIPTION
**Pull request by WalterBright**
not worth being a flag

### CODE REVIEW

**Comment by WalterBright** [comment]
Simplified the array logic, too.

**Review by RazvanN7 — APPROVED** [review]

**Review by RazvanN7 — CHANGES_REQUESTED** [review]
Hang on! You are transforming a compile time constant into a field that is allocated. This makes `VisualCPPMangler` use more memory.

File: compiler/src/dmd/cppmanglewin.d [hunk 1]
```diff
@@ -102,24 +99,24 @@ private final class VisualCPPMangler : Visitor
...
 
     alias IS_NOT_TOP_TYPE = Flags.IS_NOT_TOP_TYPE;
     alias MANGLE_RETURN_TYPE = Flags.MANGLE_RETURN_TYPE;
     alias IGNORE_CONST = Flags.IGNORE_CONST;
     alias IS_DMC = Flags.IS_DMC;
-    alias ESCAPE = Flags.ESCAPE;
 
     int flags;
+
+    bool ESCAPE;    // ESCAPE: toplevel const non-pointer types need a '$$C' escape in addition to a cv qualifier.
```
Comment (dkorpel) [inline compiler/src/dmd/cppmanglewin.d:118]:
```suggestion
    bool escape;    // ESCAPE: toplevel const non-pointer types need a '$$C' escape in addition to a cv qualifier.
```
Variable field should be camelCase

**Review by dkorpel — DISMISSED** [review]

**Comment by dkorpel** [comment]
> This makes `VisualCPPMangler` use more memory.

It's stack allocated, subtracting the stack pointer a few more bytes is not going to make a difference.

**Comment by WalterBright** [comment]
@RazvanN7 there are 5 bit flags, consuming 4 bytes, aligned on 8 bytes. Transforming them into 5 bools uses 5 bytes, aligned on 8 bytes. No difference in memory consumption. The bools also need less code to manipulate.

**Comment by RazvanN7** [comment]
Right, my point was that the enum does not allocate anything at run time. But anyway, as @dkorpel pointed out this is only stack allocated, so it doesn't matter.

**Review by RazvanN7 — APPROVED** [review]

File: compiler/src/dmd/cppmanglewin.d [hunk 2]
```diff
@@ -410,7 +405,8 @@ public:
             buf.writestring("$$A6");
         }
         buf.writestring(arg);
-        flags &= ~(IS_NOT_TOP_TYPE | IGNORE_CONST);
+        flags &= ~IS_NOT_TOP_TYPE;
+	IGNORE_CONST = false;
```
Comment (dkorpel) [inline compiler/src/dmd/cppmanglewin.d:409]: This is indented with tabs instead of spaces

File: compiler/src/dmd/cppmanglewin.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/cppmanglewin.d]: ok

**Pull request merged**